File: 66-PlusOne/66-PlusOne.java
class Solution {
    public int[] plusOne(int[] digits) {
        int i = digits.length -1;
        boolean changed = false;

        while (i >= 0){
            if (digits[i] == 9){
                digits[i] = 0;
                i--;
            }else{
                digits[i]++;
                changed = true;
                break;
            }
        }

        if (!changed){
           int[] newArr = new int[digits.length + 1];
           newArr[0] = 1;
           
           for (i=0; i < digits.length; i++){
            newArr[i+1] = digits[i];
           }
           return newArr;
        }

        return digits;


    }
}

File: 66-PlusOne/66-PlusOne.py
# Last updated: 01/01/2026, 10:21:09
1class Solution:
2    def plusOne(self, digits: List[int]) -> List[int]:
3        
4        for i in range(len(digits)-1,-1,-1):
5            if digits[i] == 9:
6                digits[i] = 0
7            else:
8                digits[i] += 1
9                return digits
10
11        return [1] + digits
12                